### PR TITLE
Add support for libgda on darwin

### PR DESCRIPTION
**Pull request by lavoiesl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
   ```
    $ /nix/store/b3sddkvy7amam6rmr4cbf423rwazm5rl-libgda-5.2.9/bin/gda-sql
    gda> .lp
           Installed providers list
    Provider  │ Description
    ──────────┼──────────────────────────────
    MySQL     │ Provider for MySQL databases
    SQLCipher │ Provider for SQLCipher
    SQLite    │ Provider for SQLite databases
   ```
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @jtojnar

### CODE REVIEW

**Comment by hedning** [comment]
@GrahamcOfBorg build libgda

File: pkgs/development/libraries/libgda/default.nix [hunk 1]
```diff
@@ -47,6 +47,6 @@ assert postgresSupport -> postgresql != null;
     homepage = https://www.gnome-db.org/;
     license = [ licenses.lgpl2 licenses.gpl2 ];
     maintainers = gnome3.maintainers;
-    platforms = platforms.linux;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (worldofpeace) [inline pkgs/development/libraries/libgda/default.nix]: not `platforms.unix`?

Comment (lavoiesl) [reply pkgs/development/libraries/libgda/default.nix]: That includes things like freebsd. I’m ok changing it, but I don't have the means to test it

File: pkgs/development/libraries/libgda/default.nix [hunk 2]
```diff
@@ -15,9 +15,17 @@ assert postgresSupport -> postgresql != null;
...
-    ++ (optional (mysqlSupport) "--with-mysql=yes")
-    ++ (optional (postgresSupport) "--with-postgres=yes");
+  configureFlags = with stdenv.lib; [
+    "--enable-gi-system-install=no"
+    "--with-mysql=${if mysqlSupport then "yes" else "no"}"
+    "--with-postgres=${if postgresSupport then "yes" else "no"}"
+
+    # macOS builds use the sqlite source code that comes with libgda,
+    # as opposed to using the system or brewed sqlite3, which is not supported on macOS,
+    # as mentioned in https://github.com/GNOME/libgda/blob/95eeca4b0470f347c645a27f714c62aa6e59f820/libgda/sqlite/README#L31
```
Comment (jtojnar) [inline pkgs/development/libraries/libgda/default.nix]: Here is the linked paper: https://web.archive.org/web/20100610151539/http://lattice.umiacs.umd.edu/files/functions_tr.pdf

**Comment by lavoiesl** [comment]
It's my first time contributing, is this PR missing anything before it can be merged?

I’m not entirely sure what it required from the checklist template.

I couldn't run `nix-shell -p nix-review --run "nix-review wip"` and I didn't figure out how to run `nix path-info -S`

**Review by hedning — APPROVED** [review]
LGTM at least.

**Comment by worldofpeace** [comment]
> It's my first time contributing, is this PR missing anything before it can be merged?
> 
> I’m not entirely sure what it required from the checklist template.
> 
> I couldn't run `nix-shell -p nix-review --run "nix-review wip"` and I didn't figure out how to run `nix path-info -S`

I'd say using the archive link from jtojnar is important.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
see @jtojnar's comment

File: pkgs/development/libraries/libgda/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/development/libraries/libgda/default.nix]: Guess it's fine.

**Comment by lavoiesl** [comment]
My bad, I read that as a footnote, not a request for adding a comment 😄 

Added 👍

**Comment by worldofpeace** [comment]
Thanks for contributing @lavoiesl :cherry_blossom: 

> I’m not entirely sure what it required from the checklist template.
> I couldn't run nix-shell -p nix-review --run "nix-review wip" and I didn't figure out how to run nix path-info -S

None of the things are required to submit a PR, it's mostly a checklist of info for the reviewers.
Only minimum thing for things to be considered is if they fit https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md. It's encouraged to check whatever you believe you've done or is valid.

As for `nix path-info -S`, you should give it the result symlink for what was built
```
nix path-info -S ./result
```
and it's only really useful info if you maybe added a dependency. And in this case I would have been interested in checking that changes for darwin don't have effects on linux.
As you've built it on darwin, I don't think you could've produced that info.

**Pull request merged**